app/models/manageiq/providers/redhat/inventory/collector.rb

```diff
@@ -0,0 +1,102 @@
+class ManageIQ::Providers::Redhat::Inventory::Collector < ManagerRefresh::Inventory::Collector
+  require_nested :InfraManager
+  require_nested :TargetCollection
+
+  attr_reader :clusters
+  attr_reader :networks
+  attr_reader :storagedomains
+  attr_reader :datacenters
+  attr_reader :hosts
+  attr_reader :vms
+  attr_reader :templates
+
+  VERSION_HASH = {:version => 4}.freeze
+
+  def initialize(manager, _target)
+    super
+
+    initialize_inventory_sources
+  end
+
+  def initialize_inventory_sources
+    @clusters       = []
+    @networks       = []
+    @storagedomains = []
+    @datacenters    = []
+    @hosts          = []
+    @vms            = []
+    @templates      = []
+  end
+
+  def hash_collection
+    # TODO: check whether needed
+    ::ManageIQ::Providers::Redhat::Inventory::HashCollection
+  end
+
+  def collect_clusters
```

Comment:
nit: we could even drop the collect_

might be that collector.clusters.each reads better than collector.collect_clusters.each